client/logic/ui/action_panel.gd

```diff
@@ -38,39 +38,17 @@ func _process(_delta: float) -> void:
 		position.y -= size.y
 
 
-func set_buttons(actions: Array) -> void:
-	if GameConst.Actions.MOVE in actions:
-		($VBoxContainer/ButtonMove as Button).show()
-	else:
-		($VBoxContainer/ButtonMove as Button).hide()
-	if GameConst.Actions.ATTACK in actions:
-		($VBoxContainer/ButtonAttack as Button).show()
-	else:
-		($VBoxContainer/ButtonAttack as Button).hide()
-	if GameConst.Actions.ENTER in actions:
-		($VBoxContainer/ButtonEnter as Button).show()
-	else:
-		($VBoxContainer/ButtonEnter as Button).hide()
-	if GameConst.Actions.DEPLOY in actions:
-		($VBoxContainer/ButtonDeploy as Button).show()
-	else:
-		($VBoxContainer/ButtonDeploy as Button).hide()
-	if GameConst.Actions.JOIN in actions:
-		($VBoxContainer/ButtonJoin as Button).show()
-	else:
-		($VBoxContainer/ButtonJoin as Button).hide()
-	if GameConst.Actions.REFILL in actions:
-		($VBoxContainer/ButtonSupply as Button).show()
-	else:
-		($VBoxContainer/ButtonSupply as Button).hide()
-	if GameConst.Actions.CAPTURE in actions:
-		($VBoxContainer/ButtonCapture as Button).show()
-	else:
-		($VBoxContainer/ButtonCapture as Button).hide()
+func set_buttons(actions: Array[GameConst.Actions]) -> void:
+	for action: GameConst.Actions in _buttons:
+		var button: Button = _buttons[action]
```

Comment:
Shouldn't we validate if the button is null? Maybe some assert

Reply:
The button can only be null if someone intentionally sets a dictionary entry to null.